crates/db_views/src/post_view.rs

```diff
@@ -392,54 +384,40 @@ fn queries<'a>() -> Queries<
         .filter(not(post::removed.or(post::deleted)));
     }
 
-    // If there is a content warning, show nsfw content by default.
-    let has_content_warning = site.content_warning.is_some();
-    if !options
-      .local_user
-      .map(|l| l.local_user.show_nsfw)
-      .unwrap_or(has_content_warning)
-    {
+    if !viewer.show_nsfw() {
       query = query
         .filter(post::nsfw.eq(false))
         .filter(community::nsfw.eq(false));
     };
 
-    if !options
-      .local_user
-      .map(|l| l.local_user.show_bot_accounts)
-      .unwrap_or(true)
-    {
+    if !viewer.show_bot_accounts() {
       query = query.filter(person::bot_account.eq(false));
     };
 
     // If its saved only, then filter, and order by the saved time, not the comment creation time.
-    if let (true, Some(_person_id)) = (options.saved_only, my_person_id) {
+    if options.saved_only {
```

Comment:
Is the person existing check being done somewhere else now?

Reply:
No, enabling `saved_only` without being logged in will now filter out all posts instead of doing nothing. That's how it normally works in views that use `left_join` instead of `exists`.